src/lib.rs

```diff
@@ -65,6 +65,44 @@ impl Build {
         }
     }
 
+    #[cfg(windows)]
+    fn check_env_var(&self, var_name: &str) -> Option<bool> {
+        env::var_os(var_name).and_then(|s| {
+            let tmp = s
+                .to_str()
+                .expect("The environment variable has invalid Unicode")
+                .to_lowercase();
+            match &tmp[..] {
+                "y" | "yes" | "on" | "true" | "t" | "1" => Some(true),
+                "n" | "no" | "off" | "false" | "f" | "-1" => Some(false),
+                "a" | "auto" | "0" => None,
+                _ => panic!(
+                    "The environment variable {} is set to an unrecognizable value: {}",
+                    var_name, tmp
+                ),
+            }
+        })
+    }
+
+    #[cfg(windows)]
+    fn is_nasm_ready(&self) -> bool {
+        self.check_env_var("OPENSSL_RUST_USE_NASM")
+            .unwrap_or_else(|| {
+                // On Windows, use cmd `where` command to check if nasm is installed
+                let wherenasm = Command::new("cmd")
+                    .args(&["/C", "where nasm"])
+                    .output()
+                    .expect("Failed to execute `cmd`.");
+                wherenasm.status.success()
+            })
+    }
+
+    #[cfg(not(windows))]
```

Comment:
I think the `#[cfg]` here can be avoided since nothing is windows-specific in that it only compiles on Windows.

Reply:
I don't think these bothers. I think this notices us that we don't consider running nasm.exe on a non-windows os. (Even though nasm has non-windows version. However, according to OpenSSL's docs, it doesn't need nasm to build on non-windows platforms.)